lib/connect.js

```diff
@@ -234,25 +240,86 @@ const getTasks = (model, setupListeners) => {
         setupListeners(mongoClient);
       }
 
+      let maybeTunnelError;
+
+      if (tunnel) {
+        tunnel.on('error', (tunnelError) => {
+          if (tunnelError.origin === 'ssh-client') {
+            maybeTunnelError = tunnelError;
+          }
+        });
+      }
+
       mongoClient.connect((err, _client) => {
+        // When mongo client failed to connect AND tunnel encountered an error
+        // those two are totally related and we want to return a tunnel error as
+        // it will have more relevant context for the end user
+        err = err && maybeTunnelError ? maybeTunnelError : err;
+
         ctx(err);
 
         if (err) {
           if (tunnel) {
             debug('data-service connection error, shutting down ssh tunnel');
-            tunnel.close();
+            tunnel.close().then(
+              () => {
+                debug('ssh tunnel stopped');
+                cb(err);
```

Comment:
`cb(null)`?

Reply:
Now i get it, sorry .. this code is quite complex though .. i think we should give the promise version / or a more de-tangled approach a chance